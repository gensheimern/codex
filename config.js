module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
