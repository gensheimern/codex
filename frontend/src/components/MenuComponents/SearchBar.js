import React from 'react';
import FlatButton from 'material-ui/FlatButton';
import IconSearch from 'material-ui/svg-icons/action/search';
import IconDelete from 'material-ui/svg-icons/navigation/close';
import IconFilter from 'material-ui/svg-icons/hardware/keyboard-arrow-down';
import TextField from 'material-ui/TextField';
import DropDownMenu from 'material-ui/DropDownMenu';
import MenuItem from 'material-ui/MenuItem';


export default class SearchBar extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
        open: false,
        value: 1,
        searchWord: '',

    };
    this.handleFilterChange = this.handleFilterChange.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  handleFilterChange = (event, index, value) =>{
    this.setState({value:value});
    console.log(value);

    switch(value.toString()) {
            case "1":
         this.props.searchFilterFeed("TimeDown","Filter");
         this.setState({value:value});
      ;break
            case "2":
         this.props.searchFilterFeed("TimeUp","Filter");
         this.setState({value:value});
      ;break
            case '3':
         this.props.searchFilterFeed("NameDown","Filter");
         this.setState({value:value});
      ;break
            case '4':
         this.props.searchFilterFeed("NameUp","Filter");
         this.setState({value:value});
      ;break
            case '5':
         this.props.searchFilterFeed("Oldest","Filter");
         this.setState({value:value});
      ;break
            case '6':
         this.props.searchFilterFeed("Newest","Filter");
         this.setState({value:value});
      ;break
        default:
        this.setState({value:value});
;
      }
    }
    handleSearch(){
      this.props.searchFilterFeed(this.state.searchWord,"Search");
    }

    handleChange = (event) => {
      console.log(event.target.value);
      this.setState({
        searchWord: event.target.value,
      });
    };

    handleClose(){

      this.props.searchFilterFeed(null,"Search");
      this.props.searchFilterFeed("TimeDown","Filter");
      this.props.changeShow();
      this.setState({
        searchWord: "",
      });
    }

render(){

  return(<div>{this.props.show &&
            <div background-color="#ccff99">
              <DropDownMenu
                 value={this.state.value}
                 onChange={this.handleFilterChange}
                 style={{ width: "30%",
                 float:"left", margin:"0%", fontSize:"8pt"}}
                 autoWidth={false}
               >
                <MenuItem value={1} primaryText="TimeDown" />
                <MenuItem value={2} primaryText="TimeUp" />
                <MenuItem value={3} primaryText="NameDown" />
                <MenuItem value={4} primaryText="NameUp" />
                <MenuItem value={5} primaryText="Oldest" />
                <MenuItem value={6} primaryText="Newest" />
              </DropDownMenu>

              <TextField
                 id="text-field-controlled"
                 value={this.state.searchWord}
                 onChange={this.handleChange}
                 style={{float:"left",margin:"0%",marginTop:"4%", width:"40%", height:"6vh"}}
                />

              <FlatButton
                icon={<IconDelete style={{
                width:"50%"}}/>}
                onClick={this.handleClose}
                style={{float:"left", width:"10%", minWidth:"15%",marginTop:"2%",}}
              />

              <FlatButton
                icon={<IconSearch style={{
                width:"50%"}}/>}
                onClick={this.handleSearch}
                style={{float:"left", width:"10%", minWidth:"15%",marginTop:"2%",}}
              />
            </div>
          }
        </div>
  )
}



}