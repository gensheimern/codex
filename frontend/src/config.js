const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;