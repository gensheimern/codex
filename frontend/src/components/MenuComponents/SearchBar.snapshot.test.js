import React from 'react';
import renderer from'react-test-renderer';
import SearchBar from './SearchBar';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const fetchMock = require('fetch-mock');

beforeEach(() => {
    window.localStorage = {getItem: () => ""};
    fetchMock.get(config.apiPath +'/user/me',{
        id: 6,
        firstName: 'Max',
        name: 'Mustermann',
        email: 'max.mustermann@gmail.com',
        image: '',
        me: true,
    });

    fetchMock.get(config.apiPath +'/user/me/notifications/unseen', {
        unseenNotifications: 4,
		
    });
});

afterEach(() => {
    fetchMock.restore();
   });

describe('SearchBar Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <MuiThemeProvider>
                    <SearchBar/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});