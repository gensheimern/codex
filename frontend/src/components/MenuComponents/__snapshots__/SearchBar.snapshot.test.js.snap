// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Snapshot renders 1`] = `
Array [
  <div
    style={
        Object {
            "float": "left",
            "height": "100%",
            "padding": 0,
            "paddingLeft": "56px",
            "width": "calc(100% - 168px)",
          }
    }
>
    <div
        className={undefined}
        style={
            Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 48,
                "lineHeight": "24px",
                "marginBottom": "15px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": "100%",
              }
        }
    >
        <input
            autoFocus={true}
            disabled={false}
            id="SearchBarTextField"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            style={
                Object {
                    "WebkitOpacity": 1,
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "appearance": "textfield",
                    "backgroundColor": "rgba(0,0,0,0)",
                    "border": "none",
                    "color": "white",
                    "cursor": "inherit",
                    "font": "inherit",
                    "height": "100%",
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
            }
            type="text"
            value=""
        />
        <div>
            <hr
                aria-hidden="true"
                style={
                    Object {
                        "borderBottomStyle": "solid",
                        "borderBottomWidth": 1,
                        "borderColor": "#e0e0e0",
                        "borderLeft": "none",
                        "borderRight": "none",
                        "borderTop": "none",
                        "bottom": 8,
                        "boxSizing": "content-box",
                        "margin": 0,
                        "muiPrepared": true,
                        "position": "absolute",
                        "width": "100%",
                      }
                }
            />
            <hr
                aria-hidden="true"
                style={
                    Object {
                        "borderBottom": "none",
                        "borderBottomStyle": "solid",
                        "borderBottomWidth": 2,
                        "borderColor": "#00bcd4",
                        "borderLeft": "none",
                        "borderRight": "none",
                        "borderTop": "none",
                        "bottom": 8,
                        "boxSizing": "content-box",
                        "margin": 0,
                        "muiPrepared": true,
                        "position": "absolute",
                        "transform": "scaleX(0)",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "width": "100%",
                      }
                }
            />
        </div>
    </div>
</div>,
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    style={
        Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "float": "left",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": "100%",
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": "56px",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "width": "56px",
          }
    }
    tabIndex={0}
    type="button"
>
    <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
    >
        <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
                Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "white",
                    "height": 24,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "verticalAlign": "middle",
                    "width": 24,
                  }
            }
            viewBox="0 0 24 24"
        >
            <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
        </svg>
    </div>
</button>,
]
`;
